[BUG] Unable to generate API Token

**Describe the bug**
On latest version, I am unable to generate an API token.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to user API Token Management
2. Click on Generate API Token
3. An error message “Internal Server Error” is displayed
4. In the logs, I see the message:
[ERROR] Error generating API token: SqliteError: table api_tokens has no column named name


**Environment (please complete the following information):**
* Host OS: Linux
* Deployment type: Docker
  * latest, 3.10.1
* Database backend: SQLite

--- a/src/db/migrations.test.ts
+++ b/src/db/migrations.test.ts
@@ -2,8 +2,8 @@ import { describe, it, expect } from 'vitest';
 import { registry } from './migrations.js';
 
 describe('migrations registry', () => {
-  it('has all 16 migrations registered', () => {
-    expect(registry.count()).toBe(16);
+  it('has all 17 migrations registered', () => {
+    expect(registry.count()).toBe(17);
   });
 
   it('first migration is v37 baseline', () => {
@@ -12,14 +12,14 @@ describe('migrations registry', () => {
     expect(all[0].name).toContain('v37_baseline');
   });
 
-  it('last migration is the system backup column rename', () => {
+  it('last migration is the api_tokens name column', () => {
     const all = registry.getAll();
     const last = all[all.length - 1];
-    expect(last.number).toBe(16);
-    expect(last.name).toContain('rename_system_backup_columns');
+    expect(last.number).toBe(17);
+    expect(last.name).toContain('api_tokens_name');
   });
 
-  it('migrations are sequentially numbered from 1 to 16', () => {
+  it('migrations are sequentially numbered from 1 to 17', () => {
     const all = registry.getAll();
     for (let i = 0; i < all.length; i++) {
       expect(all[i].number).toBe(i + 1);

--- a/src/db/migrations.ts
+++ b/src/db/migrations.ts
@@ -1,7 +1,7 @@
 /**
  * Migration Registry Barrel File
  *
- * Registers all 16 migrations in sequential order for use by the migration runner.
+ * Registers all 17 migrations in sequential order for use by the migration runner.
  * Migration 001 is the v3.7 baseline (selfIdempotent — handles its own detection).
  * Migrations 002-011 were originally 078-087 and retain their original settingsKeys
  * for upgrade compatibility.
@@ -28,6 +28,7 @@ import { migration as auditLogColumnsMigration, runMigration013Postgres, runMigr
 import { migration as messagesDecryptedByMigration, runMigration014Postgres, runMigration014Mysql } from '../server/migrations/014_add_messages_decrypted_by.js';
 import { migration as notificationPrefsUniqueMigration, runMigration015Postgres, runMigration015Mysql } from '../server/migrations/015_add_notification_prefs_unique.js';
 import { migration as renameSystemBackupColumnsMigration, runMigration016Postgres, runMigration016Mysql } from '../server/migrations/016_rename_system_backup_columns.js';
+import { migration as apiTokensNameMigration, runMigration017Postgres, runMigration017Mysql } from '../server/migrations/017_add_api_tokens_name_column.js';
 
 // ============================================================================
 // Registry
@@ -215,4 +216,19 @@ registry.register({
   sqlite: (db) => renameSystemBackupColumnsMigration.up(db),
   postgres: (client) => runMigration016Postgres(client),
   mysql: (pool) => runMigration016Mysql(pool),
+});
+
+// ---------------------------------------------------------------------------
+// Migration 017: Add missing name and expires_at columns to api_tokens
+// Pre-3.7 databases created api_tokens without these columns.
+// Fixes: https://github.com/Yeraze/meshmonitor/issues/2435
+// ---------------------------------------------------------------------------
+
+registry.register({
+  number: 17,
+  name: 'add_api_tokens_name_column',
+  settingsKey: 'migration_017_add_api_tokens_name_column',
+  sqlite: (db) => apiTokensNameMigration.up(db),
+  postgres: (client) => runMigration017Postgres(client),
+  mysql: (pool) => runMigration017Mysql(pool),
 });

--- a/src/server/migrations/017_add_api_tokens_name_column.ts
+++ b/src/server/migrations/017_add_api_tokens_name_column.ts
@@ -1,0 +1,101 @@
+/**
+ * Migration 017: Add missing columns to api_tokens table
+ *
+ * Pre-3.7 databases created api_tokens without the `name` and `expires_at`
+ * columns. The v3.7 baseline uses CREATE TABLE IF NOT EXISTS, which doesn't
+ * alter existing tables. This migration adds the missing columns.
+ *
+ * Fixes: https://github.com/Yeraze/meshmonitor/issues/2435
+ */
+import type { Database } from 'better-sqlite3';
+import { logger } from '../../utils/logger.js';
+
+// ============ SQLite ============
+
+export const migration = {
+  up: (db: Database): void => {
+    logger.info('Running migration 017 (SQLite): Adding missing api_tokens columns...');
+
+    // Add name column (default to 'API Token' for existing rows)
+    try {
+      db.exec("ALTER TABLE api_tokens ADD COLUMN name TEXT NOT NULL DEFAULT 'API Token'");
+      logger.debug('Added name column to api_tokens');
+    } catch (e: any) {
+      if (e.message?.includes('duplicate column')) {
+        logger.debug('api_tokens.name already exists, skipping');
+      } else {
+        logger.warn('Could not add name to api_tokens:', e.message);
+      }
+    }
+
+    // Add expires_at column
+    try {
+      db.exec('ALTER TABLE api_tokens ADD COLUMN expires_at INTEGER');
+      logger.debug('Added expires_at column to api_tokens');
+    } catch (e: any) {
+      if (e.message?.includes('duplicate column')) {
+        logger.debug('api_tokens.expires_at already exists, skipping');
+      } else {
+        logger.warn('Could not add expires_at to api_tokens:', e.message);
+      }
+    }
+
+    logger.info('Migration 017 complete (SQLite): api_tokens columns aligned');
+  },
+
+  down: (_db: Database): void => {
+    logger.debug('Migration 017 down: Not implemented (destructive column drops)');
+  }
+};
+
+// ============ PostgreSQL ============
+
+export async function runMigration017Postgres(client: import('pg').PoolClient): Promise<void> {
+  logger.info('Running migration 017 (PostgreSQL): Ensuring api_tokens columns exist...');
+
+  try {
+    await client.query("ALTER TABLE api_tokens ADD COLUMN IF NOT EXISTS name TEXT NOT NULL DEFAULT 'API Token'");
+    await client.query('ALTER TABLE api_tokens ADD COLUMN IF NOT EXISTS "expiresAt" BIGINT');
+    logger.debug('Ensured name/expiresAt exist on api_tokens');
+  } catch (error: any) {
+    logger.error('Migration 017 (PostgreSQL) failed:', error.message);
+    throw error;
+  }
+
+  logger.info('Migration 017 complete (PostgreSQL): api_tokens columns aligned');
+}
+
+// ============ MySQL ============
+
+export async function runMigration017Mysql(pool: import('mysql2/promise').Pool): Promise<void> {
+  logger.info('Running migration 017 (MySQL): Ensuring api_tokens columns exist...');
+
+  try {
+    const [nameRows] = await pool.query(`
+      SELECT COLUMN_NAME FROM information_schema.COLUMNS
+      WHERE TABLE_SCHEMA = DATABASE() AND TABLE_NAME = 'api_tokens' AND COLUMN_NAME = 'name'
+    `);
+    if (!Array.isArray(nameRows) || nameRows.length === 0) {
+      await pool.query("ALTER TABLE api_tokens ADD COLUMN name VARCHAR(255) NOT NULL DEFAULT 'API Token'");
+      logger.debug('Added name to api_tokens');
+    } else {
+      logger.debug('api_tokens.name already exists, skipping');
+    }
+
+    const [expiresRows] = await pool.query(`
+      SELECT COLUMN_NAME FROM information_schema.COLUMNS
+      WHERE TABLE_SCHEMA = DATABASE() AND TABLE_NAME = 'api_tokens' AND COLUMN_NAME = 'expiresAt'
+    `);
+    if (!Array.isArray(expiresRows) || expiresRows.length === 0) {
+      await pool.query('ALTER TABLE api_tokens ADD COLUMN expiresAt BIGINT');
+      logger.debug('Added expiresAt to api_tokens');
+    } else {
+      logger.debug('api_tokens.expiresAt already exists, skipping');
+    }
+  } catch (error: any) {
+    logger.error('Migration 017 (MySQL) failed:', error.message);
+    throw error;
+  }
+
+  logger.info('Migration 017 complete (MySQL): api_tokens columns aligned');
+}

--- a/src/server/models/APIToken.test.ts
+++ b/src/server/models/APIToken.test.ts
@@ -61,11 +61,13 @@ const createTestDatabase = () => {
         CREATE TABLE IF NOT EXISTS api_tokens (
           id INTEGER PRIMARY KEY AUTOINCREMENT,
           user_id INTEGER NOT NULL,
+          name TEXT NOT NULL DEFAULT 'API Token',
           token_hash TEXT UNIQUE NOT NULL,
           prefix TEXT NOT NULL,
           is_active INTEGER NOT NULL DEFAULT 1,
           created_at INTEGER NOT NULL,
           last_used_at INTEGER,
+          expires_at INTEGER,
           created_by INTEGER NOT NULL,
           revoked_at INTEGER,
           revoked_by INTEGER,

